ViAn/Test/filehandlertest.cpp

```diff
@@ -4,76 +4,257 @@
+
+    QVERIFY(file_handler->proj_equals(*proj1,*l_proj1));
+    QVERIFY(file_handler->proj_equals(*proj2,*l_proj2));
+    QVERIFY(file_handler->proj_equals(*proj3,*l_proj3));
+
+    QVERIFY(file_handler->delete_project(proj1->id));
+    QVERIFY(file_handler->delete_project(proj2->id));
+    QVERIFY(file_handler->delete_project(proj3->id));
+}
+
 /**
- * @brief FileHandlerTest::projectHandlingTest
- * Tests project creation and deletion.
- * TODO : tests project write to file and read from file
+ * @brief FileHandlerTest::project_add_remove_items
+ * Tests adding and removing videos from project, also
+ * test information is saved and loaded correctly.
  */
-void filehandlertest::project_handling_test(){
-    FileHandler* fh = new FileHandler();
-    Project* proj =  fh->create_project("TEST_PROJ");
-    std::string v1 = "video1.txt"; // names for testfiles
-    std::string v2 = "video2.txt";
-    std::string v3 = "video3.txt";
-    ID vid1 = fh->create_file(v1, proj->files->dir);  // create testfiles
-    ID vid2 = fh->create_file(v2, proj->files->dir);
-    ID vid3 = fh->create_file(v3, proj->files->dir);
-    fh->add_video(proj, fh->get_file(vid1)); //add tesfiles as videos to project.
-    fh->add_video(proj, fh->get_file(vid2));
-    fh->add_video(proj, fh->get_file(vid3));
+void FileHandlerTest::project_add_remove_items(){
+    Project* proj1 = file_handler->create_project("TEST_PROJ1");
+    Project* proj2 = file_handler->create_project("TEST_PROJ2");
+    Project* proj3 = file_handler->create_project("TEST_PROJ3");
 
-    fh->save_project(proj);
-    //check file contents
-    Project* proj2 = fh->load_project("TEST_PROJ", fh->work_space + "TEST_PROJ");
-    QVERIFY(fh->proj_equals(*proj2,*proj));
+    file_handler->add_video(proj1, "TEST_VID1");
+    file_handler->add_video(proj2, "TEST_VID2");
+    file_handler->add_video(proj3, "TEST_VID3");
 
-    //check project contents
-    fh->delete_file(vid1);
-    fh->delete_file(vid2);
-    fh->delete_file(vid3);
-    QCOMPARE(fh->delete_project(proj->id), 0);
+    file_handler->save_project(proj1->id);
+    file_handler->save_project(proj2->id);
+    file_handler->save_project(proj3->id);
+
+
+    Project* l_proj1 = file_handler->load_project(file_handler->get_dir(proj1->dir).absoluteFilePath("TEST_PROJ1.json").toStdString());
+    Project* l_proj2 = file_handler->load_project(file_handler->get_dir(proj2->dir).absoluteFilePath("TEST_PROJ2.json").toStdString());
+    Project* l_proj3 = file_handler->load_project(file_handler->get_dir(proj3->dir).absoluteFilePath("TEST_PROJ3.json").toStdString());
+
+    QVERIFY(file_handler->proj_equals(*proj1,*l_proj1));
+    QVERIFY(file_handler->proj_equals(*proj2,*l_proj2));
+    QVERIFY(file_handler->proj_equals(*proj3,*l_proj3));
+
+    QVERIFY(file_handler->delete_project(proj1->id));
+    QVERIFY(file_handler->delete_project(proj2->id));
+    QVERIFY(file_handler->delete_project(proj3->id));
```

Comment:
Same comment here about delete project. 
There should be a test where you add multiple videos to one project as well, to make sure multiple videos works when saving, loading and deleting projects.